# Assignment 1 - ANLY 535
### Date: 11/28/2021
### Rushabh Barbhaya

In [1]:
import numpy as np

W1 is 2x5
First layer W = 2x5 and b = 5x1

In [2]:
np.random.seed(seed=289)
_input = np.array([[0.3], [0.4]])
W1 = np.random.rand(2, 5)
b1 = np.random.rand(5, 1)
b1

array([[0.48966863],
       [0.61544854],
       [0.62467448],
       [0.08738796],
       [0.90462271]])

In [3]:
z1 = np.dot(W1.transpose(), _input)+b1
z1

array([[0.89019087],
       [0.93947114],
       [0.84754806],
       [0.35762942],
       [1.39493903]])

sigmoid = 1/(1-(e^-z))

In [4]:
a1 = 1/(1+np.exp(-z1))
a1

array([[0.70892956],
       [0.71899282],
       [0.70005254],
       [0.58846646],
       [0.80137956]])

W2 is 1x5
b2 is 1x1

In [5]:
np.random.seed(seed=289)
W2 = np.random.rand(5, 1)
b2 = np.random.rand(1, 1)
z2 = np.dot(W2.transpose(), a1)+b2
z2

array([[2.6055998]])

In [6]:
a2 = 1/(1+np.exp(-z2))
a2

array([[0.9312211]])

#### Following the lecture notes for the logic to this assignment. Two sensor inputs are 0.3 and 0.4 and with the seed set, we get the output as 0.93.
#### So there is 93% chance of flood